src/osd/PrimaryLogPG.cc

```diff
@@ -6471,6 +6504,30 @@ int PrimaryLogPG::do_osd_ops(OpContext *ctx, vector<OSDOp>& ops)
       ++ctx->num_write;
       tracepoint(osd, do_osd_op_pre_delete, soid.oid.name.c_str(), soid.snap.val);
       {
+	if (oi.has_manifest()) {
+	  if (oi.flags & object_info_t::FLAG_HAS_REFERENCE && oi.manifest.is_redirect()) {
+	      object_locator_t target_oloc(oi.manifest.redirect_target);
+	      refcount_manifest(ctx->obc, target_oloc, oi.manifest.redirect_target, 
+				SnapContext(), "chunk_put", NULL, 0);
+
+	  } else if (oi.manifest.is_chunked()) {
+	    for (auto p : oi.manifest.chunk_map) {
+	      if (p.second.flags & chunk_info_t::FLAG_HAS_REFERENCE) {
+		object_locator_t target_oloc(p.second.oid);
+		refcount_manifest(ctx->obc, target_oloc, p.second.oid, 
+				  SnapContext(), "chunk_put", NULL, p.first);
+	      }
+	    }
+	  } 
+	} else {
+	  cls_chunk_refcount_read_ret read_ret;
+	  int ref = read_refcount_manifest(ctx, read_ret);
```

Comment:
Hmm, I don't think we should interfere with delete here.  If someone goes and sends delete ops to a rados pool with refcounted objects that's their mistake.

If we're worried about it, perhaps a pool property that disallows delete ops, or caps that only allow read operations and the refcount cls.

Reply:
You are right. Fixed.